src/Storages/StorageMergeTree.cpp

```diff
@@ -379,8 +379,19 @@ void StorageMergeTree::alter(
     if (!query_settings[Setting::allow_suspicious_primary_key])
         MergeTreeData::verifySortingKey(new_metadata.sorting_key);
 
+
+    bool indices_changed = false;
+
+    if (new_metadata.getSettingsChanges())
+    {
+        const auto current_changes = new_metadata.getSettingsChanges()->as<const ASTSetQuery &>().changes;
+        auto find_setting = std::find_if(current_changes.begin(), current_changes.end(), [](const auto & change){ return change.name == "enable_minmax_index_for_all_numeric_columns"; });
```

Comment:
Hmm, why does this not check the corresponding setting for string columns?

Reply:
it was supposed to be removed when I made setting immutable, removed it now, thanks